tools/lsp/server_loop.rs

```diff
@@ -483,64 +458,81 @@ pub async fn set_binding_command(
         }
     }?;
 
-    if let Some(element) = element_at_position(document_cache, &uri, &element_range.start) {
-        if let Some(node) = &element.borrow().node {
-            let (start, end) = {
-                let mut offset_to_position = document_cache.offset_to_position_mapper(&uri);
-                let r = node.text_range();
-                (offset_to_position.map(r.start().into()), offset_to_position.map(r.end().into()))
-            };
-
-            if start != element_range.start {
-                return Err(format!(
-                    "Element found, but does not start at the expected place (){start:?} != {:?}).",
-                    element_range.start
-                )
-                .into());
-            }
-            if end != element_range.end {
-                return Err(format!(
-                    "Element found, but does not end at the expected place (){end:?} != {:?}).",
-                    element_range.end
-                )
-                .into());
-            }
-
-            let (result, edit) = properties::set_binding(
-                document_cache,
-                &uri,
-                &element,
-                &property_name,
-                new_expression,
-            )?;
-
-            if !dry_run {
-                if let Some(edit) = edit {
-                    let response = notifier
-                        .send_request::<lsp_types::request::ApplyWorkspaceEdit>(
-                            lsp_types::ApplyWorkspaceEditParams {
-                                label: Some("set binding".into()),
-                                edit,
-                            },
-                        )?
-                        .await?;
-                    if !response.applied {
-                        return Err(response
-                            .failure_reason
-                            .unwrap_or("Operation failed, no specific reason given".into())
-                            .into());
-                    }
+    let (result, edit) = {
+        let document_cache = &mut ctx.document_cache.borrow_mut();
+        let uri = text_document.uri;
+        if let Some(source_version) = text_document.version {
+            if let Some(current_version) = document_cache.document_version(&uri) {
+                if current_version != source_version {
+                    return Err(
+                        "Document version mismatch. Please refresh your property information"
```

Comment:
Not specific for this commit, but this error is a bit strange. "Please refresh your property information"?  How do i do that as the user seeing this error message?